From aa4b15d0b26b3c72fab736bcd28a67ab9d1404b7 Mon Sep 17 00:00:00 2001
From: John Sallay <jasallay@gmail.com>
Date: Sat, 23 Oct 2021 08:26:42 -0400
Subject: [PATCH] Fix issue 4595 qa_agc Assertion Error.

The number of input elements needs to be disivible by volk_alignment, which it wasn't
for machines with 512-bit registers.

Signed-off-by: John Sallay <jasallay@gmail.com>
---
 gr-analog/python/analog/qa_agc.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gr-analog/python/analog/qa_agc.py b/gr-analog/python/analog/qa_agc.py
index a5a8ea47b6a..9368dd5d2a8 100644
--- a/gr-analog/python/analog/qa_agc.py
+++ b/gr-analog/python/analog/qa_agc.py
@@ -454,7 +454,9 @@ def test_006(self):
         tb = self.tb
 
         sampling_freq = 100
-        N = int(5*sampling_freq)
+        # N must by a multiple of the volk_alignment of the system for this test to work.
+        # For a machine with 512-bit registers, that would be 8 complex-floats.
+        N = int(8 * sampling_freq)
         src1 = analog.sig_source_c(sampling_freq, analog.GR_SIN_WAVE,
                                    sampling_freq * 0.10, 100)
         dst1 = blocks.vector_sink_c()
